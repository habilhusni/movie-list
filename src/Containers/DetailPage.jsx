import React from "react";
import { connect } from "react-redux";
import PropTypes from "prop-types";
import Detail from "../Components/Detail";
import Loading from "../Components/Loading";

const DetailPage = (props) =>
  props.details.isLoading ? <Loading /> : <Detail details={props.details} />;

DetailPage.propTypes = {
  details: PropTypes.object,
};

DetailPage.defaultProps = {
  details: {},
};

const mapStateToProps = (state) => ({
  details: state.details,
});

const mapDispatchToProps = (dispatch) => ({});

export default connect(mapStateToProps, mapDispatchToProps)(DetailPage);
