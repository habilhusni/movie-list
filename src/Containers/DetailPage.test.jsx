import React from "react";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import toJSON from "enzyme-to-json";
import store from "../common/mockStore";

import DetailPage from "./DetailPage";

Enzyme.configure({ adapter: new Adapter() });

describe("Test Snapshot and Functional", () => {
  let component;

  beforeEach(() => {
    component = shallow(<DetailPage store={store} />);
  });

  it("Testing DetailPage Snapshot", () => {
    expect(toJSON(component)).toMatchSnapshot();
  });
});
