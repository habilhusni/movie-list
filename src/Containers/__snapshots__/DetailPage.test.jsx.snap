// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Snapshot and Functional Testing DetailPage Snapshot 1`] = `
<ContextProvider
  value={null}
>
  <DetailPage
    details={
      Object {
        "error": Object {},
        "isLoading": false,
        "payload": Object {},
      }
    }
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  />
</ContextProvider>
`;
